salt/fileserver/hgfs.py

```diff
@@ -314,7 +314,7 @@ def init():
     if new_remote:
         remote_map = os.path.join(__opts__['cachedir'], 'hgfs/remote_map.txt')
         try:
-            with salt.utils.fopen(remote_map, 'w+') as fp_:
+            with salt.utils.fopen(remote_map, 'wb+') as fp_:
                 timestamp = datetime.now().strftime('%d %b %Y %H:%M:%S.%f')
                 fp_.write('# hgfs_remote map as of {0}\n'.format(timestamp))
```

Comment:
Since you now opened in binary mode, you can only write bytes to the file. Since this is ASCII, it's safe to do `b'ascii string'`. On py3 of course.